### PR TITLE
[TEST] cache module metadata

### PR DESCRIPTION
**Pull request by zekun000**

### CODE REVIEW

File: aptos-move/aptos-vm/src/move_vm_ext/vm.rs [hunk 1]
```diff
@@ -63,10 +72,26 @@ impl MoveVmExt {
...
+            ret.clone()
+        } else {
+            drop(read);
+            let ret = Arc::new(aptos_framework::get_vm_metadata(
+                &self.inner,
+                module_id.clone(),
+            ));
+            self.cache.write().unwrap().insert(module_id, ret.clone());
+            ret
+        }
```
Comment (davidiw) [inline aptos-move/aptos-vm/src/move_vm_ext/vm.rs:92]: can we verify that a previous write hasn't already done this work?
seems like a better approach might be to get a write lock, check if it doesn't exist, set it if it doesn't, then perform a read -- of course then you have the yuckiness of handling an impossible get that is conditional though it should have been set...

**Review by davidiw — COMMENTED** [review]
I guess this addresses the eventual optimization needed for handling repeated queries in the execution side. I wonder what type of cache we should consider. For example, should we just try to cache everything? Do we want an LRU? Do we need to make it capable of handling attacks.

**Comment by zekun000** [comment]
ideally the metadata cache should be binded with the module cache itself, that'd be part of the new loader work though.

File: aptos-move/aptos-vm/src/move_vm_ext/vm.rs (hunk 1 above)
Comment (zekun000) [reply aptos-move/aptos-vm/src/move_vm_ext/vm.rs:92]: I don't think it'd make big difference if I understand what you suggested here, currently VM is not accessed concurrently

**Pull request closed without merging**